lib/Parser/Parse.cpp

```diff
@@ -12749,15 +12819,30 @@ ParseNodePtr Parser::ParseDestructuredVarDecl(tokens declarationType, bool isDec
             }
         }
 
-        if (m_token.tk != tkID && m_token.tk != tkTHIS && m_token.tk != tkSUPER && m_token.tk != tkLCurly && m_token.tk != tkLBrack)
+        
+        if (m_token.tk != tkID && m_token.tk != tkTHIS && m_token.tk != tkSUPER)
         {
-            if (isDecl)
+            if (isObjectPattern && (m_token.tk == tkLCurly || m_token.tk == tkLBrack))
```

Comment:
This conditional and the next seem very similar, and took me a moment to parse the difference. I think it would be clearer if you gave a name to the token subexpression and used that. Hopefully you can give a better name than I, but maybe something like this:

```cpp
bool tokenStartsObject = m_token.tk == tkLCurly || m_token.tk == tkLBrack;
if ((isObjectPattern && tokenStartsObject) || (!isObjectPattern && !tokenStartsObject) {
  …
```